header-rewrite-filter/header_processor.cc

```diff
@@ -32,20 +32,18 @@ namespace HeaderRewriteFilter {
 
         // parse values and call setVals
         try {
-            std::vector<std::string> vals;
-            for(auto it = start + 1; it != operation_expression.end(); ++it) {
-                if (*it == "if") { // condition found
+            std::string val;
+            val = std::string{*(start + 1)};
```

Comment:
Nit: could be a single line, which would also let it be const and avoid the copy from the `operator=`:
`const std::string val(*(start + 1));`